### ISSUE TITLE
v5 new component: App Logo (Top Bar)

### ISSUE THREAD

**Issue by nurm717123** [reapit/elements#225]
**Abstract**

_As part of the v5 Elements release, the ticket #153 was created. To break it into smaller part and allow for more focused discussions, this ticket has been created._

***app bar (parent)***
![image](https://github.com/user-attachments/assets/dbc50742-49e5-4c0d-aebf-2305aa72bfc8)

***this ticket***
![image](https://github.com/user-attachments/assets/a211f99e-cd27-40f2-a57a-7f35adc98717)

**Context**
- the App Logo is a component that would be displayed as part of the Top Bar
- App Logo could differ depending on which app will be using (e.g Reapit, Agentbox, or other Reapit product)
- following [this](https://github.com/reapit/elements/issues/153#issuecomment-2487897635) discussion it would need separate discussion from top bar

As discussed in #153, this ticket is for creating reusable Logo component to be used inside `TopBar` component

**Specification**

- Is new component or existing? new
- Is the component being re-named? No
- Is the component design signed off by product and design? yes, as part of App Bar / Top Bar ticket
- [Guide to component structure & documentation in storybook](https://github.com/reapit/elements/wiki/Developer-Contribution-Guide)
- [Link to DS in Figma](https://www.figma.com/design/6CaivqdlTX0UkFYJkpBKDu/Reapit-DS?node-id=1564-7620&node-type=symbol&m=dev)
- [Link to Elements Audit in Figma](https://www.figma.com/design/XJ6qcAV8gHscsUodqJMNEF/App-bar-and-Side-nav-%5BElements%5D?node-id=18-7578&t=AB2erFDHYuLLxvGj-4)
- [Accessibility requirements](https://m3.material.io/components/top-app-bar/accessibility)
- [Link to Design Guidelines](https://www.figma.com/design/pJ96bxoEFStjjIbAPy0Jaj/Design-Guidelines?node-id=945-25707&t=mvIlOCZjDYP90TZT-4)
- [Parent ticket](https://github.com/reapit/elements/issues/153)

**Developer Checklist**

- [ ] Styles alignment between Design System and Elements
- [ ] Check design tokens in Figma and implement CSS variable tokens if available for relevant component
- [ ] Align with accessibility standards / spec as per above
- [ ] If relevant, break down component into `Styles Only` and `React` component structures
- [ ] Ensure all variants of components are documented as appropriate 
- [ ] Ensure unit test coverage is adequate for component
- [ ] Update documentation in MDX file as per guidelines

**Release Checklist**

- [ ] Approved PR merged to main
- [ ] Design & product review and feedback addressed by developer
- [ ] Beta release by product / engineering lead to next beta version

**Additional Context or Information**

**Comment by nurm717123** [comment]
Based on the context described above, my plan is to provide a component with the following interface:

```html
<!-- html user -->
<a href="/"  classname="el-top-bar-logo">
  <img src="your.svg"/> <!-- could be .png too -->
</a>
```

```jsx
// react consumer
<TopBar>
 <TopBar.Logo /> {/* use default Reapit logo and "/" href */}
  {/* use custom src, href, or callback */}
 <TopBar.Logo src="custom.svg" href="/home" />
 <TopBar.Logo src={customBrandIcon} onClick={(e) => {
     e.preventDefault();
     navigate('/home');
  }} />
</TopBar>
```

If there is any input or feedback, it would be really appreciated.

**Comment by kurtdoherty** [comment]
@nurm717123 Thanks for the proposal 👏  A few comments:

**thought:** I'm not sure I love the idea of this component dealing with both the layout of the brand logo _and_ the display of the logo itself. Currently, all products will incur a (slight) increase in bundle size due to the default inclusion of a generic Reapit logo, despite the fact very few of them will need such a logo.

**thought:** Currently, all our other TopBar compound components just take care of the layout concern (which is really good 👌). Your proposal here for `TopBar.Logo` is doing more than just the layout. This means consumers are constrained to displaying their logo via a `src` URL instead of being free to choose how they display it (e.g. child React component that renders inline SVG vs child `<img />` like your HTML-only example demonstrates, and so on).

**suggestion:** I don't think a default logo should be baked into `TopBar.Logo`. If we need to provide a generic Reapit brand component (not sure we really do), Elements could provide one by exporting a React component and/or inline data URI. 

**suggestion:** I think `TopBar.Logo` should accept `children` like the rest of the `TopBar` compound components.

To summarise, I think `TopBar.Logo` should simply take care of the `<a>` element you show in the HTML only example. The `<img>` (or inline SVG, or ...) that the consumer needs to display for the actual logo can be provided via `children`.

**Comment by nurm717123** [comment]
thank you @kurtdoherty for the feedback 🙏.

> thought: I'm not sure I love the idea of this component dealing with both the layout of the brand logo and the display of the logo itself. With this approach, all products would incur a (slight) increase in bundle size due to the default inclusion of a generic Reapit logo, despite the fact very few of them will need such a logo.

I hadn't considered that before; that's a great insight 👍 

> thought: Currently, all our other TopBar compound components just take care of the layout concern (which is really good 👌). Your proposal here for TopBar.Logo is doing more than just the layout. This means consumers are constrained to displaying their logo via a src URL instead of being free to choose how they display it (e.g. child React component that renders inline SVG vs child <img /> like your HTML-only example demonstrates, and so on).

Yes, I thought standardizing the use of the `img` tag was a good idea at first. But then I realized that HTML users might not follow that rule anyway. And like you said, since the other `TopBar` components just handle the layout, it makes sense to keep it consistent with them. That really does make sense 👍 

> suggestion: I think TopBar.Logo should accept children like the rest of the TopBar compound components.

Definitely


Great 😊 , now that I think that its clear to me, my conclusion is to make the new component `TopBar.Logo` just as a wrapper to give the layout styling for React consumer .eg:

```javascript
<TopBar.Logo> {/* padding, margin, display, .etc */}
   {/* consumer can use children themself as the other TopBar pattern */}
   <BrandIcon />
    {/* or */}
    <img src="brand.png" />
</TopBar.Logo>
```

Should I go ahead and submit a PR now?

**Comment by kurtdoherty** [comment]
Yep, that looks good. Also, although I've provided some feedback that's resulted in changes, I think you approached the API design quite well here. I appreciated the thought that went into the different ways you could use you TopBar.Logo component 👏

**Comment by kurtdoherty** [comment]
Closed by #229
